### PR TITLE
Refactor: use STL algorithms (2)

### PR DESCRIPTION
**Pull request by angriman**
As in #869: use STL algorithms where applicable.

### CODE REVIEW

File: networkit/cpp/distance/NeighborhoodFunction.cpp [hunk 1]
```diff
@@ -46,9 +46,10 @@ void NeighborhoodFunction::run() {
             }
         }
     }
-    for (index i = 1; i < size-1; ++i) {
-        result[i] += result[i-1];
-    }
+
+    if (result.size() > 2)
+        std::partial_sum(result.begin(), result.end() - 1, result.begin());
```
Comment (avdgrinten) [inline networkit/cpp/distance/NeighborhoodFunction.cpp:51]: I think the `- 1` is broken here. The previous code operated on the whole array but this replacement skips the last element.

File: networkit/cpp/distance/Volume.cpp [hunk 2]
```diff
@@ -51,21 +51,14 @@ std::vector<double> Volume::volume(const Graph &G, std::vector<double> rs, count
...
+                return nodeDist.second <= rs[i];
+            });
     }
     std::vector<double> ys;
-    ys.reserve(xs.size());
-    for (const auto &x : xs) {
-        ys.push_back(x / samples);
-    }
+    std::transform(xs.cbegin(), xs.cend(), std::back_inserter(ys),
+                   [samples](double x) { return x / static_cast<double>(samples); });
```
Comment (avdgrinten) [inline networkit/cpp/distance/Volume.cpp:61]: Not sure if this is really simpler than the original code. Anyway, we should keep the `.reserve()`.

File: networkit/cpp/distance/NeighborhoodFunction.cpp (hunk 1 above)
Comment (angriman) [reply networkit/cpp/distance/NeighborhoodFunction.cpp:51]: Ah true, the vector has size `size - 1`.

**Review by avdgrinten — APPROVED** [review]

**Review by fabratu — APPROVED** [review]
LGTM.

**Pull request merged**